### ISSUE TITLE
Window closing didn't trigger Combine transform on Dataflow Runner using Python SDK

### ISSUE THREAD

**Issue by BjornPrime** [apache/beam#24682]
### What happened?

In the course of debugging another issue on Dataflow Runner, it was noticed that the Combine stage of the below pipeline only triggered on pipeline draining. The windows seemed to begin and end correctly, that is, they each contained four sequential elements, but they were not processed through the Combine stage until draining began.

The pipeline in question:
```
import logging

import apache_beam as beam
from apache_beam.options.pipeline_options import PipelineOptions
from apache_beam.transforms import trigger
from apache_beam.transforms import window
from apache_beam.transforms.periodicsequence import PeriodicImpulse
import time


def run(argv=None):
  options = PipelineOptions()
  pipeline = beam.Pipeline(options=options)
  _ = (
      pipeline
      | PeriodicImpulse(
          start_timestamp=time.time(),
          stop_timestamp=time.time() + 15,
          fire_interval=1,
          apply_windowing=False)
      | beam.WindowInto(
          window.GlobalWindows(),
          trigger=trigger.Repeatedly(trigger.AfterCount(4)),
          accumulation_mode=trigger.AccumulationMode.DISCARDING)
      | beam.combiners.Top.Largest(1)
      | beam.ParDo(lambda x: logging.error('element: %s', x))
  )
  pipeline.run().wait_until_finish()


if __name__ == '__main__':
  logging.getLogger().setLevel(logging.INFO)
  run()
```

### Issue Priority

Priority: 2 (default / most bugs should be filed as P2)

### Issue Components

- [X] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [X] Component: Google Cloud Dataflow Runner

**Comment by BjornPrime** [comment]
.remove-labels go

**Comment by damccorm** [comment]
This should be fixed by https://github.com/apache/beam/pull/23507 when it is released with 2.44.0
